src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
 
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (elementData.length == size) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     public T peek() {
-        return null;
+        if (empty() == true) {
+            throw new EmptyStackException();
+        }
+        return elementData[size - 1];
     }
 
     public T pop() {
-        return null;
+        if (empty() == true) {
+            throw new EmptyStackException();
+        }
```

Comment:
same

Reply:
Call this only in peek as you call peek on line 32 and use only empty() without == true as empty() returns boolean

Reply:
done